src/testdir/test_listdict.vim

```diff
@@ -950,13 +960,22 @@ func Test_reduce()
 
   call assert_fails("call reduce([], { acc, val -> acc + val })", 'E998: Reduce of an empty List with no initial value')
   call assert_fails("call reduce(0z, { acc, val -> acc + val })", 'E998: Reduce of an empty Blob with no initial value')
+  call assert_fails("call reduce('', { acc, val -> acc + val })", 'E998: Reduce of an empty String with no initial value')
 
-  call assert_fails("call reduce({}, { acc, val -> acc + val }, 1)", 'E897:')
-  call assert_fails("call reduce(0, { acc, val -> acc + val }, 1)", 'E897:')
-  call assert_fails("call reduce('', { acc, val -> acc + val }, 1)", 'E897:')
+  call assert_fails("call reduce({}, { acc, val -> acc + val }, 1)", 'E1098:')
+  call assert_fails("call reduce(0, { acc, val -> acc + val }, 1)", 'E1098:')
   call assert_fails("call reduce([1, 2], 'Xdoes_not_exist')", 'E117:')
   call assert_fails("echo reduce(0z01, { acc, val -> 2 * acc + val }, '')", 'E39:')
 
+  call assert_fails("vim9 reduce(0, (acc, val) => (acc .. val), '')", 'E9998:')
+  call assert_fails("vim9 reduce({}, (acc, val) => (acc .. val), '')", 'E9998:')
+  call assert_fails("vim9 reduce(0.1, (acc, val) => (acc .. val), '')", 'E9998:')
+  call assert_fails("vim9 reduce(function('tr'), (acc, val) => (acc .. val), '')", 'E9998:')
+  call assert_fails("call reduce('', { acc, val -> acc + val }, 1)", 'E9999:')
+  call assert_fails("call reduce('', { acc, val -> acc + val }, {})", 'E9999:')
+  call assert_fails("call reduce('', { acc, val -> acc + val }, 0.1)", 'E9999:')
+  call assert_fails("call reduce('', { acc, val -> acc + val }, function('tr'))", 'E9999:')
+
```

Comment:
Can you also add a test for using test_null_string() as the first argument to reduce()?

Reply:
Thank you for checking. I'll add the tests.